lib/components/conversation/conversation_item.dart

```diff
@@ -0,0 +1,163 @@
+          _onSelectController.sink.add(_id);
+        } else {
+          logger.warning("No listener for ConversationItemView.onSelect");
+        }
+      });
+
+    var idElement = DivElement()
+      ..className = "conversation-item-id"
+      ..innerText = _id;
+
+    var messageElement = DivElement()..className = "conversation-item-message";
+    var messageTextElement = SpanElement()
+      ..className = "conversation-item-message-text"
+      ..innerText = _message;
+    _messageStatusElement = SpanElement()..className = "conversation-item-status";
+    _updateStatus(_status);
+
+    messageElement..append(messageTextElement)..append(_messageStatusElement);
+    contentWrapper..append(idElement)..append(messageElement);
+    renderElement..append(checkboxWrapper)..append(contentWrapper);
+
+    this._onSelectController = StreamController();
+    this._onSelect = _onSelectController.stream;
+    this._onCheckController = StreamController();
+    this._onCheck = _onCheckController.stream;
+    this._onUncheckController = StreamController();
+    this._onUncheck = _onUncheckController.stream;
+  }
+
+  void _updateStatus(ConversationItemStatus status) {
+    var messageStatusClasses = _messageStatusElement.classes
+      ..removeWhere((className) => className.startsWith("converversation-item-status--"));
+    var renderElementClasses = renderElement.classes
+      ..removeWhere((classname) => !classname.startsWith("conversation-item--selected") && classname.startsWith("conversation-item--"));
+    switch (status) {
+      case ConversationItemStatus.draft:
+        renderElement.classes.add("conversation-item--draft");
+        messageStatusClasses.add("converversation-item-status--draft");
+        _messageStatusElement
+          ..innerText = "[draft]"
+          ..className = messageStatusClasses.join(" ");
+        break;
+      case ConversationItemStatus.failed:
+        renderElementClasses.add("conversation-item--failed");
+        renderElement.className = renderElementClasses.join(" ");
+        messageStatusClasses.add("converversation-item-status--failed");
+        _messageStatusElement
+          ..innerText = "[delivery failure]"
+          ..className = messageStatusClasses.join(" ");
+        break;
+      case ConversationItemStatus.pending:
+        renderElementClasses.add("conversation-item--pending");
+        renderElement.className = renderElementClasses.join(" ");
+        messageStatusClasses.add("converversation-item-status--pending");
+        _messageStatusElement
+          ..innerText = "[pending]"
+          ..className = messageStatusClasses.join(" ");
+        break;
+      default:
+        renderElementClasses.add("conversation-item--normal");
+        renderElement.className = renderElementClasses.join(" ");
+        messageStatusClasses.add("converversation-item-status--normal");
+        _messageStatusElement
+          ..innerText = ""
+          ..className = messageStatusClasses.join(" ");
```

Comment:
nit:
```suggestion
        renderElement.classes.add("conversation-item--normal");
        _messageStatusElement.classes.add("converversation-item-status--normal");
        _messageStatusElement.innerText = "";
        break;
```

I know adding the `break` is not actually doing anything, but it avoids mistakes in the future in case we turn the default into a normal case.